app/src/main/java/com/example/vshcheglov/webshop/ui/adapters/ProductsRecyclerAdapter.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.vshcheglov.webshop.ui.adapters
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.example.vshcheglov.webshop.R
+import com.example.vshcheglov.webshop.domain.Product
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.product_recycler_item.view.*
+
+class ProductsRecyclerAdapter(private val productList: List<Product>)
+    : RecyclerView.Adapter<ProductsRecyclerAdapter.ViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.product_recycler_item, parent, false)
+
+        return ViewHolder(view)
+    }
+
+    override fun getItemCount() = productList.size
+
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bindProduct(productList[position])
+    }
+
+    class ViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
+        fun bindProduct(product: Product) {
```

Comment:
Please don't do in a such way. ViewHolder has another responsibility. Execute this code in `onBindViewHolder `